### PR TITLE
feat(urlTools) : expose all url tools and more in a dedicated module

### PR DESCRIPTION
**Pull request by bobylito**
- new namespace `url` for all functions related to query string serialization deserialization
 - move static function to the new module
 - added a new function to serialize any `SearchParameters`
 - getState filter feature now exposed as filter method on `SearchParameters`

### CODE REVIEW

**Comment by redox** [comment]
Nice :)

I would go for a few renamings:
 - `getConfigurationFromQueryString -> getStateFromQueryString`
 - `getForeignConfigurationInQueryString -> getUnrecognizedParametersInQueryString`
 - not yet sure about `urlTools`, maybe I would go for just `url`

**Comment by bobylito** [comment]
All done @redox!

File: README.md [hunk 1]
```diff
@@ -684,15 +684,21 @@ helper.getState(['query', 'attribute:*']);
...
 ```js
-var state = algoliasearchHelper.AlgoliaSearchHelper.getConfigurationFromQueryString(qs);
+var state = algoliasearchHelper.url.getStateFromQueryString(qs);
 ```
 
 #### Get the configuration contained in the query string that is not for the helper
 
 ```js
-var otherConf = algoliasearchHelper.AlgoliaSearchHelper.getForeignConfigurationInQueryString(qs);
+var otherConf = algoliasearchHelper.url.getUnrecognizedParametersInQueryString(qs);
```
Comment (vvo) [inline README.md]: `getExtraParametersInQueryString`?

File: test/spec/helper.state.js [hunk 2]
```diff
@@ -361,7 +361,7 @@ test('getConfigurationFromQueryString should parse page as number and be consist
 
   var queryString = helper.getStateAsQueryString({filters: filters});
 
-  var partialStateFromQueryString = algoliasearchHelper.AlgoliaSearchHelper.getConfigurationFromQueryString(
```
Comment (vvo) [inline test/spec/helper.state.js]: Do we have a check that ensure that the previous `algoliasearchHelper.AlgoliaSearchHelper.` methods are still there and works?

Comment (bobylito) [inline test/spec/helper.state.js]: no

File: README.md (hunk 1 above)
Comment (bobylito) [inline README.md]: Not bad. @redox did a little research yesterday on the topic and it seems that unrecognized should be pretty common.

Comment (vvo) [inline README.md]: Ok I was not sure myself but if unrecognized is native-compliant then allright. Unrecognized gives more detail about what we are searching for, `extra` is generic indeed.

File: test/spec/helper.state.js (hunk 2 above)
Comment (vvo) [inline test/spec/helper.state.js]: Maybe we can add one so that no developer will remove it (BUT when we do a V3 for example).

Comment (vvo) [inline test/spec/helper.state.js]: Maybe we can add one so that no developer will remove it (BUT when we do a V3 for example).

**Comment by bobylito** [comment]
Ok will add some compat tests :)

**Comment by bobylito** [comment]
I added some compat tests and updated the doc. We're good to go, @vvo ?

File: test/spec/compat.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+var test = require('tape');
+var algoliaSearchHelper = require('../../index');
+var url = algoliaSearchHelper.url;
+var AlgoliaSearchHelper = algoliaSearchHelper.AlgoliaSearchHelper;
+
+test('Should be compatible', function(t) {
+  t.equal(
+    AlgoliaSearchHelper.getConfigurationFromQueryString,
```
Comment (vvo) [inline test/spec/compat.js]: ahah nice trick :)

**Comment by vvo** [comment]
:fireworks: go

**Pull request merged**